compiler/psi/src/org/jetbrains/kotlin/parsing/KotlinParsing.java

```diff
@@ -885,12 +895,10 @@ assert _at(IDENTIFIER) ||
 
         PsiBuilder.Marker reference = mark();
         PsiBuilder.Marker typeReference = mark();
-        parseUserType(/* allowSimpleIntersectionTypes */ false);
+        parseUserType(/* allowSimpleIntersectionTypes */ false,  /* strictTypeArguments */ true);
         typeReference.done(TYPE_REFERENCE);
         reference.done(CONSTRUCTOR_CALLEE);
 
-        parseTypeArgumentList();
```

Comment:
While it seemed incorrect, I think it's better to be done in a separate commit with a linked youtrack issue that should be structured in accordance with guidelines: https://kotlinfoundation.org/submitting-incompatible-changes/

But the simplest thing we could do here is just postponing removing that line yet :)

Reply:
I removed it from the PR and I'll report it accordingly.